nightshift/datastore.py

```diff
@@ -24,12 +22,59 @@
 )
 from sqlalchemy.dialects.postgresql import ENUM
 from sqlalchemy.ext.declarative import declarative_base
-from sqlalchemy.orm import relationship
+from sqlalchemy.orm import relationship, sessionmaker
 
 
 Base = declarative_base()
 
 
+def conf_db():
+    """
+    Retrieves db configuation from env variables
+
+    Returns:
+        db settings as dictionary
+    """
+    return dict(
+        NS_DBUSER=os.getenv("NS_DBUSER"),
+        NS_DBPASSW=os.getenv("NS_DBPASSW"),
+        NS_DBHOST=os.getenv("NS_DBHOST"),
+        NS_DBPORT=os.getenv("NS_DBPORT"),
+        NS_DBNAME=os.getenv("NS_DBNAME"),
+    )
+
+
+class DataAccessLayer:
+    def __init__(self):
+        db = conf_db()
+        self.conn = f"postgresql+psycopg2://{db['NS_DBUSER']}:{db['NS_DBPASSW']}@{db['NS_DBHOST']}:{db['NS_DBPORT']}/{db['NS_DBNAME']}"
+        self.engine = None
+
+    def connect(self):
+        self.engine = create_engine(self.conn)
+        Base.metadata.create_all(self.engine)
+        self.Session = sessionmaker(bind=self.engine)
+
+
+dal = DataAccessLayer()
+
+
+@contextmanager
+def session_scope():
+    """
+    Provides a transactional scope around series of operations.
+    """
+    dal.connect()
+    session = dal.Session()
+    try:
+        yield session
+    except:
+        session.rollback()
+        raise
+    finally:
```

Comment:
Would it be a good or bad idea to automatically run `session.commit()` here? Unsure of how granular you want the commits to be. Could also provide a backstop to ensure that no changes go uncommitted to the db.

Reply:
I've been thinking about it but could not decide one way or the other. I think there might be situations when it would be better to commit records before session ends, but that can be done explicitly mid session too. I will add commit before session closes.